[CLD-7567] Deprecate Self Serve: Second Pass

nickmisasi opened the pull request:
#### Summary
This PR includes a more extensive clean up of various code related to Cloud and Self Hosted Self Serve purchases. This PR also includes the removal of Self Serve true up review submissions through the system console to the CWS. 

This PR _does not_ include the removal of limits, and likely misses some mentions of Cloud Free. I plan to address these in future PRs. 

Related PR in Enterprise repo: https://github.com/mattermost/enterprise/pull/1672

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-7567

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
Removed support for self serve true up review submission in system console
```

nickmisasi commented:
/update-branch

nickmisasi commented:
/update-branch

mattermost-build commented:
Error trying to update the PR.
Please do it manually.

mattermost-build commented:
E2E tests not automatically triggered, because the PR is not in a mergeable state. Please update the branch with the base branch and resolve outstanding conflicts.

nickmisasi commented:
cc @hmhealey since package.json changed - I removed the Stripe dependencies as they're no longer necessary.

nickmisasi commented:
/update-branch

marianunez commented on webapp/channels/src/components/admin_console/admin_definition.tsx:
```diff
@@ -327,13 +327,7 @@ const AdminDefinition: AdminDefinitionType = {
-                it.not(it.licensedForFeature('Cloud')),
-                it.configIsFalse('ServiceSettings', 'SelfHostedPurchase'),
-            ),
+            it.not(it.licensedForFeature('Cloud')),
```
nit: we may not need that `it.any` anymore

marianunez commented on webapp/channels/src/components/admin_console/admin_definition.tsx:
```diff
@@ -357,6 +351,7 @@ const AdminDefinition: AdminDefinitionType = {
                     id: 'BillingHistory',
                     component: BillingHistory,
                 },
+                isHidden: it.not(it.licensedForFeature('Cloud')),
```
This may not be needed here if the parent already checks for this?

marianunez commented on webapp/channels/src/components/admin_console/billing/billing_summary/upsell_card.scss:
```diff
@@ -1,69 +0,0 @@
-@import 'utils/mixins';
-
-.UpsellCard {
```
We don't need this anymore? Seems we still have the `UpsellCard` component

nickmisasi replied:
UpsellCard isn't used anywhere, must have missed deleting the component itself

marianunez commented on webapp/channels/src/components/admin_console/feature_discovery/feature_discovery.tsx:
```diff
@@ -169,32 +162,22 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
-                    <FeatureDiscoveryCloudStartTrialButton
-                        telemetryId={`start_cloud_trial_from_${this.props.featureName}`}
-                        extraClass='btn btn-primary'
+            // if all conditions are set for being able to request a cloud trial, then show the cta start cloud trial button
```
We can probably delete this comment as it's outdated re: cloud trial

marianunez commented on webapp/channels/src/components/admin_console/feature_discovery/feature_discovery.tsx:
```diff
@@ -169,32 +162,22 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
+                    className='btn btn-primary'
+                    data-testid='featureDiscovery_primaryCallToAction'
+                    onClick={() => {
+                        trackEvent(TELEMETRY_CATEGORIES.SELF_HOSTED_ADMIN, 'click_enterprise_contact_sales_feature_discovery');
```
This is pre-existing but seems to be labeled in the wrong telemetry category if this is a cloud case?

marianunez commented on webapp/channels/src/components/admin_console/feature_discovery/feature_discovery.tsx:
```diff
@@ -212,62 +195,35 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
-                        />
-                    )}
-                </p>}
+                {(!this.props.isCloud) && (<p className='trial-legal-terms'>
```
Not sure if this `gettingTrialError` is referring to getting a cloud trial or self-hosted trial. If it's cloud we can probably get rid of that code too below

marianunez commented on webapp/channels/src/packages/mattermost-redux/src/action_types/hosted_customer.ts:
```diff
@@ -6,15 +6,5 @@ import keyMirror from 'mattermost-redux/utils/key_mirror';
 export default keyMirror({
     SELF_HOSTED_PRODUCTS_REQUEST: null,
```
Do we still need this self hosted get products ?

nickmisasi replied:
Yes, for the pricing modal

marianunez reviewed (approved):
Thanks Nick! Code removals overall LGTM - some non-blocking comments below. 

We want to make sure to test that the self-hosted trial requests and renewal prompts with Contact Sales still works as expected after this cleanup.

yasserfaraazkhan reviewed (commented):
- when on enterprise trial
<img width="593" alt="Screenshot 2024-04-30 at 1 23 25 PM" src="https://github.com/mattermost/mattermost/assets/15979783/f3400ac8-9321-42ca-9af3-5f3729046422">

<img width="1448" alt="Screenshot 2024-04-30 at 1 16 23 PM" src="https://github.com/mattermost/mattermost/assets/15979783/2cbbb29a-9554-4be7-bde2-28fcd087812d">

- UI issue: the contact sales is not appearing as button

<img width="341" alt="Screenshot 2024-04-30 at 1 24 49 PM" src="https://github.com/mattermost/mattermost/assets/15979783/79266676-2e3e-4e58-b324-387b2b44b002">

- tested with free instance, cloud instance, uploading license, checked when instance license expires.

Please tell me if there's any scenario we might want to cover.

yasserfaraazkhan commented:
/e2e-tests

mattermost-build commented:
Successfully triggered E2E testing!
[GitLab pipeline](http://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/423677) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/423677)

nickmisasi commented:
@yasserfaraazkhan in latest diff 
I've removed the duplicate CTAs in the banner you posted, making "Contact Sales" the primary CTA
![image](https://github.com/mattermost/mattermost/assets/12176405/f8ca5db2-6b6b-436f-8593-cbc9087cd20f)

I've fixed the spacing of the "or contact sales" in the pricing modal
![image](https://github.com/mattermost/mattermost/assets/12176405/3337ea8f-4a8e-4794-84b6-e02d8fcba2a9)

I've made the enterprise section a primary button
![image](https://github.com/mattermost/mattermost/assets/12176405/7a48d1e6-39c6-4da1-9c9d-39919fb18e24)

hmhealey commented on server/channels/store/sqlstore/store.go:
```diff
@@ -109,7 +109,6 @@ type SqlStoreStores struct {
 	postPriority               store.PostPriorityStore
 	postAcknowledgement        store.PostAcknowledgementStore
 	postPersistentNotification store.PostPersistentNotificationStore
-	trueUpReview               store.TrueUpReviewStore
```
Are there any DB migrations that need to be done to remove this table or recreate it if someone were to downgrade?

nickmisasi replied:
Good call, added!

hmhealey commented on server/public/model/client4.go:
```diff
@@ -8305,50 +8297,6 @@ func (c *Client4) GetMyIP(ctx context.Context) (*GetIPAddressResponse, *Response
 	return response, BuildResponse(r), nil
 }
 
-func (c *Client4) CreateCustomerPayment(ctx context.Context) (*StripeSetupIntent, *Response, error) {
```
Not that I expect that these to be used by anyone else, but removing these methods is technically a breaking change. I'm 0/5 though because I know I tend to be overly cautious about breaking this interfaces.

Worst case, we could always stub them out instead of calling the APIs which no longer exist

agnivade reviewed (approved):
Schema changes look good.
